### ISSUE TITLE
Login on paypal.com broken on Firefox

### ISSUE THREAD

**Issue by LeGiTiM** [bitwarden/browser#982]
Hi there!

**Description of the bug:**
When attempting to log in on paypal.com, bitwarden can fill the email, then when hitting "next", the page returns an error "Entrez le code CAPTCHA correct." (English: "Enter the correct CAPTCHA code"). [There is no visible captcha, it's a hidden procedure of the website, it always worked without captcha for the user before the bug]. Paypal refuses to go further, it's thus impossible to enter the password, and log in successfully.
(Moreover: side minor bug: the "Auto-fill feature on page load" doesn't work on paypal.com. I guess it's related to the high security standards and unsual login method, I don't care if this bug is not resolved: hitting the keyboard shortcut or the extension click-fill feature both work).

**Actions that skirt the bug:**
- disable the bitwarden extension
- being logged out/locked (vault closed) beforehand

**Tweak that doesn't do anything:**
- disable "Enable Auto-fill On Page Load" (I thought this was the reason at first, that's why I tested it first).

**Versions of Firefox tested:**
- Nightly (70.0a1)
- Firefox regular ( 68.0.1)
(I'm sorry, I don't know exactly since when (or which version) this bug is there, i'd say a couple of months?)

**Operating System:**
Windows 10 v. 1903.

**Exclusions:**
- no bug on Chromium-variant browsers
- not this bug on Microsoft Edge.

Thank you in advance for looking into it!

**Comment by kspearrin** [comment]
I am not able to reproduce. When I go to paypal.com to login in, I see email and password on the same page/form. Is this not what you see?

![image](https://user-images.githubusercontent.com/1190944/61997107-bd45c900-b06a-11e9-8ae1-c6e46391b9dc.png)

**Comment by taylorsilva** [comment]
Came here to report this bug as well. I'm using Firefox 69.0b7 with macOS.

@kspearrin I think @LeGiTiM is seeing this login screen which gives me the captcha issue as he's described:

<img width="606" alt="Screen Shot 2019-07-27 at 12 35 02 PM" src="https://user-images.githubusercontent.com/4583837/61997122-0007a100-b06b-11e9-94fa-af15a9454f2e.png">

Which appears to be the default login screen if you haven't logged in previously. If you open the paypal login page while in incognito mode you'll get this screen. After logging in and out I now see the same screen that you see and autofill works.

<img width="511" alt="Screen Shot 2019-07-27 at 12 36 26 PM" src="https://user-images.githubusercontent.com/4583837/61997138-3e9d5b80-b06b-11e9-99d2-8d9478e6e335.png">

**Comment by LeGiTiM** [comment]
Indeed, it's just as @taylorsilva explained. It happens only for a "clean visit" of the website.
(thank you btw for the tip of manually typing the mail, Taylor; it'll save me from having to lock out/disable the extension each time.)
Bitwarden is probably filling the password field in the background, and clicking "Next" somehow triggers the final login step, while the catcha validation is supposed to happen when you click "next", before you enter the password => error message.
I don't know if you can fix this bug since it seems more related to the site method itself, than the fault of bitwarden.
But maybe that trying to detect sites who use 2 steps logins like this and injecting the username, then the password would be useful? I know that Tumblr and Yahoo use a similar method (with a next button before entering the password), but I didn't get issues with bitwarden on them.

**Comment by taylorsilva** [comment]
Is there a way that bitwarden can only fill in fields that are visible on the screen? Just throwing the obvious idea out there, I'm sure "visible" is a tough state to define if the site is not explicitly using `type=hidden`.

**Comment by Christilut** [comment]
Came here to report the same problem.

Happens when I go to paypal.com and I'm not already logged in. I press CTRL-SHIFT-L and Enter and I get the captcha error message.

Firefox 68

**Comment by danielmatsumoto** [comment]
I was here to talk about the same problem
hope this is fixed :)

**Comment by kah0922** [comment]
I can confirm that I also suffer from this issues on the latest Firefox Nightly.

**Comment by SharkyRawr** [comment]
This is still an issue when using Bitwarden auto-fill. :/

**Comment by kspearrin** [comment]
Should be fixed next version.

**Comment by taylorsilva** [comment]
Thanks @kspearrin! I'm guessing this was the fix? 
https://github.com/bitwarden/browser/commit/c99403255dc4e64a7e54bb0ea86b77ad57d8c676

**Comment by kspearrin** [comment]
Yes.

**Comment by hrcl** [comment]
@kspearrin Thank you for this fix. When will this get into a new release?

**Comment by truthsword** [comment]
Similar issue w/Edge Chrome (dev). Instead of captcha, we see "Try entering the code again."

**Comment by targodan** [comment]
(Still) having the same issue on Firefox 69.0.1 64bit on ArchLinux. This is quite bad as PayPal is quite an essential website.

Any ETA on the release?

**Comment by sorryusernameisalreadytaken** [comment]
I think it is possible to fix this by adding the fields:
`login_email`
&
`login_password`

Can anybody proof this?
See my attachment.
![bitwarden_paypal_addon_problem](https://user-images.githubusercontent.com/12913436/66365427-31ff8280-e98d-11e9-93bb-fcda2d461fcd.JPG)

But it seems to be important to disable "**Enable Auto-fill On Page Load**". Otherwise some functions did not work anymore, like adding an amount to payout to your bank account.

**Comment by Hammerfest** [comment]
"Try entering the code again."

Same issue for me, been about 2 months, I just turned off autofill for paypal so that's good enough, manually typing out works fine. Now I search for it as I was annoyed this evening and look, a bug report!

**Comment by LaurentOngaro** [comment]
Same issue with vivaldi and chrome browsers with bitwarden extension installed
The solution is to open paypal in a private windows (as it, bitwarden is disabled) and paypal will valid its captcha

**Comment by mathsen** [comment]
Same issue with Firefox and Bitwarden. In private mode it works - I didn't suspect bitwarden and found this issue by accident. Please fix this.

**Comment by Tryer101** [comment]
Same here, latest Firefox and Bitwarden.

Glad I came across this thread.

**Comment by Gamertech** [comment]
Got this today on Firefox 70.0.1

**Comment by LeGiTiM** [comment]
I'm not very familiar with github, I don't know if/how I can 'reopen' this thread as an active issue. I'm afraid that the contributors/developers of bitwarden don't see/watch this one anymore. Apparently @kspearrin has done something to fix it (thank you!), but it hasn't worked, given all the reports here (still not working for me either).
I suggest newcomers with the same bug to open a new thread with description of the issue and eventually a link to this one, to just make sure they realize it's still bugged.
Have a good day everyone!

**Comment by Tryer101** [comment]
@LeGiTiM :)

I contacted Bitwarden via the official website earlier this morning with a Ref. to this thread.
I understand the Dev knows about this bug/problem but it's still occurring after many months.
Could this be due to the fact that this software is developed and maintained by only one person?
I'm contemplating the idea of switching to another password manager at this point.

**Comment by LeGiTiM** [comment]
It's true that it's annoying (especially for people who care about their privacy and wipe their cookies & local data between sessions), but it's a very specific bug to how paypal handle logins, more than the very own fault of Bitwarden's developer. I guess time is needed to find the accurate trick to get around procedures like this.
It'd be a shame to get rid of bitwarden just for this annoyance, especially when we can still login by entering manually the email with our keyboard to skirt the bug.
The problem is that this bug will affect all users of bitwarden with a fresh install of firefox, or after a cleaning of cookies/local storage, and they won't know the trick above, so reports will keep coming, or some people, who don't know the tip and won't have the reflex/patience to seek for a solution, will really ditch bitwarden because they will assume they can't login to such an essential site as paypal.
I'm sure a developer will find the solution for this at some point.

**Comment by Tryer101** [comment]
Good point! :)

One thing though, I wonder why Bitwarden's Dev don't send an email to all users so the seeking for a solution wouldn't have to be. At this age of communication, such a task is very easy to do.

**Comment by sorryusernameisalreadytaken** [comment]
@kspearrin Thank for the fix. Can you estimate when we can calculate with an rls?

**Comment by Hammerfest** [comment]
Last browser release was Jul 30, has apparently been fixed for a while, but no release date mentioned sooooooo 

We sit, we wait, my paltry renewal occurred and I am still ok with that.

**Comment by mathsen** [comment]
I really can't understand why such an important fix doesn't get released for the masses... I mean even if it means just cherry-picking  this single commit and release a hotfix for the current version. This bug probably affects much more people than those who are registered on github and search for it and comment here. Many people might even not even consider, that it's bitwarden's fault (as me for a couple of time). So please release this overdue fix.

**Comment by dlorde** [comment]
I'm getting this invisible captcha problem with PayPal in all browsers with Bitwarden extension installed. The quickest workaround I've found is to type the email address manually.

**Comment by Sigri44** [comment]
Hi ! I use Firefox & Bitwarden, and same error with auto-fill.

![paypal](https://user-images.githubusercontent.com/9304947/70867011-c8918680-1f70-11ea-8503-96d627025311.png)

**Comment by ItsPedoBear** [comment]
Same error with extension today
But work if typing manually

OS : W10
Browser : Firefox 71.0
Bitwarden : 1.41.0

**Comment by koopernick** [comment]
I have this issue for the past 3 months or so.
I couldn't find what was causing it, and now I finally find the culprit after getting angry at that error message xD

I'm having this on Firefox and Chrome (latest build Version 79.0.3945.79).

**Comment by mathsen** [comment]
This problem was the cause for me to switch too [KeePassXC](https://keepassxc.org/). Benefit: I have the database now stored on my own devices.

**Comment by truthsword** [comment]
> This problem was the cause for me to switch too KeePassXC

Still an issue here. Quite annoying. However, KeePass isn't a solution as it won't auto-type on Paypal for me either.

> Benefit: I have the database now stored on my own devices.

You can run BW on your own server.

**Comment by DanHakimi** [comment]
Please stop closing the issue, I just ran into it today.

**Comment by UtopiaOpera** [comment]
Yup... can confirm this issue still exists.

**Comment by matteocontrini** [comment]
See https://github.com/bitwarden/browser/issues/1068#issuecomment-558152893.

**Comment by truthsword** [comment]
> See [#1068 (comment)](https://github.com/bitwarden/browser/issues/1068#issuecomment-558152893).

Has the "upcoming release" happened? If so, what version number. I'm still seeing this problem. ICBW, but it seems premature to close a topic based on an "upcoming release" when users haven't confirmed the fix. Eh?

**Comment by matteocontrini** [comment]
> Has the "upcoming release" happened?

It doesn't seem so: https://github.com/bitwarden/browser/releases

> it seems premature to close a topic based on an "upcoming release"

It's actually normal on open source projects. When a fix has been implemented the issue is marked as closed in order to focus development on open issues

**Comment by hrcl** [comment]
> Has the "upcoming release" happened? If so, what version number. I'm still seeing this problem. 

current master branch is at [1.42.1](https://github.com/bitwarden/browser/blob/master/src/manifest.json), which contains the fix.
Unfortunately the add-on currently shipped for Firefox is 1.41.0. 
So either we have to 
- wait for a roll-out of a new release in Firefox's "add-on store"
- check out the options from the original reporter (https://github.com/bitwarden/browser/issues/982#issue-473512930)
- build and install it on our own, as described in the [README](https://github.com/bitwarden/browser/blob/master/README.md) (which failed for me for unknown reasons, but YMMV)

**Comment by targodan** [comment]
So... why is it not rolled out on the add-on store yet? Oversight? Limitations from the store?

**Comment by truthsword** [comment]
> So... why is it not rolled out on the add-on store yet? Oversight? Limitations from the store?

If it's any consolation,  the Chrome Store extension is at v1.41 too. So we wait.

**Comment by trassiorf** [comment]
Same error on Google Chrome for Linux with auto-fill
![Capture du 2019-12-20 12-04-19](https://user-images.githubusercontent.com/10478846/71250887-1582ad80-2321-11ea-8a46-cb67aea8a9a6.png)

**Comment by sorryusernameisalreadytaken** [comment]
> build and install it on our own, as described in the [README](https://github.com/bitwarden/browser/blob/master/README.md) (which failed for me for unknown reasons, but YMMV)

Can you explain your "unknown reasons" a bit more?

**Comment by setyb** [comment]
> So... why is it not rolled out on the add-on store yet? Oversight? Limitations from the store?

There is currently no release built since July 30 2019 in https://github.com/bitwarden/browser/releases . The fix was implemented after that date.
Thus, and it's just my guess based on the above, I doubt the code with the fix has been sent to the extension repositories yet.

**Comment by setyb** [comment]
Update: Version 1.42.2 has been built, which includes this fix.  According to Kyle, it's been sent to Mozilla, so now it's up to them to approve it and make it available.  Even if it gets approved right away, Firefox 72 will be released on the 7th of January, so bitwarden v1.42.2 will either be released before then, or be held until at least the 9th of January, because IIRC extension updates are disabled for a day (or two) during Firefox releases to reduce the possibility of compounded problems.

**Comment by LeGiTiM** [comment]
Thank you for the info @setyb !

**Comment by setyb** [comment]
> Thank you for the info @setyb !

You're welcome.  My pleasure.

**Comment by truthsword** [comment]
Hopefully, v1.42.2 has come to Firefox already, but Chrome still awaits the update.

**Comment by Tryer101** [comment]
I've just tried and it works perfectly! :)

**Comment by setyb** [comment]
@LeGiTiM You may want to update the original post now that the fix has been released and confirmed to work.

**Comment by LeGiTiM** [comment]
The ticket is closed, I think it's sufficient.
But I edited anyway for people who dig too deep ^^

**Comment by bpsib** [comment]
This problem has now resurfaced yet support can't seem to be able to reproduce it.

**Comment by brettjenkins** [comment]
Getting this myself this past week

**Comment by Mavarius1989** [comment]
indeed. Just added TOTP for Paypal and had the error.

**Comment by thetic** [comment]
this is affecting me, too

**Comment by MrPistachios** [comment]
Im getting this issue constantly

**Comment by Nibestase** [comment]
Same for me :-(
Issue on a Win11 Pro 64, Firefox 115.0.3 (64 bits), Bitwarde extension 2023.7.1, and same with Edge 115.0.1901.203 (64 bits)

![image](https://github.com/bitwarden/clients/assets/142178809/01827af5-9828-4d80-9cfd-dd973011ecb2)

![image](https://github.com/bitwarden/clients/assets/142178809/515df09d-d2d0-4c55-9498-0e89aa629403)

![image](https://github.com/bitwarden/clients/assets/142178809/c5f3d733-90df-41e3-9f89-4399a578ad7c)

**Comment by SVNKoch** [comment]
Can confirm, getting the same error.
@kspearrin could you consider reopening this ticket please?

**Comment by dlorde** [comment]
Yup, I'm getting this autofill error on PayPal too - but if I copy/paste the password, it's OK.

**Comment by bpsib** [comment]
A workaround I found that works is to remove the Authenticator key from the PayPal Bitwarden item and have it in a separate item entry. I can then use the PayPal item for username and password to autofill then the PayPal authenticator key separate item for copy and pasting the TOTP verification code.

I really think there should be an option in Bitwarden to turn off attempted TOTP auto-filling in certain items to avoid problems like this in the future.

**Comment by cagonzalezcs** [comment]
@bpsib 

That idea is certainly something I can bring up with the team.

With that in mind, this issue presents in a similar manner to https://github.com/bitwarden/clients/issues/5880 and has been resolved with the merge of PR https://github.com/bitwarden/clients/pull/5700 . The changes from that PR are set to release with v2023.8.

The closing comment on https://github.com/bitwarden/clients/issues/5880 provides more context on why the issue existed and how it was fixed.

**Comment by ackstorm23** [comment]
This is not fixed, or you have additional bugs causing teh same problem.  This is still happening for me and results in me having to use a different browser without Bitwarden to make ANY paypal purchases.

**Comment by bpsib** [comment]
@ackstorm23 It was broken again by [v2023.7.0](https://github.com/bitwarden/clients/releases/tag/browser-v2023.7.0) and has sice been fixed again by [v2023.8.2](https://github.com/bitwarden/clients/releases/tag/browser-v2023.8.2) which should be updated on Firefox very soon.

**Comment by barthermans** [comment]
@cagonzalezcs any idea when v2023.8 will become available in the Firefox extensions store?

**Comment by cagonzalezcs** [comment]
@barthermans 

The update should be rolling out fairly soon, though we did recently have some issues arise with regards to submission of the extension on Firefox. 

Our devops team made some improvements to simplify and optimize the bundle submission process to the various web extension storefronts. Unfortunately, a small issue with how that build process was compiling the source files caused a delay for the release.

That said, we worked out the issues yesterday and have resubmitted the extension to the Firefox storefront. I apologize for the inconvenience.
